server/models/commands/ChangeDataVersionCommand.py

```diff
@@ -20,17 +30,66 @@ def backward_impl(self):
         self.wf_module.save()
         self.wf_module.set_fetched_data_version(self.old_version)
 
+    async def schedule_execute(self) -> None:
+        """
+        Tell renderers to render the new workflow, _maybe_.
+
+        ChangeDataVersionCommand is often created from a fetch, and fetches
+        are often invoked by cron. These tend to be our most resource-heavy
+        operations: e.g., Twitter-accumulate with 1M records. So let's use lazy
+        rendering.
+
+        From our point of view:
+
+            * If self.workflow has notifications, render.
+            * If anybody is viewing self.workflow right now, render.
+
+        Of course, it's impossible for us to know whether anybody is viewing
+        self.workflow. So we _broadcast_ to them and ask _them_ to request a
+        render if they're listening. This gives N render requests (one per
+        Websockets communicator) instead of 1, but we don't mind because
+        spurious render requests are no-ops.
+
+        From the user's point of view:
+
+            * If I'm viewing self.workflow, changing data versions causes a
+              render. (There isn't even any HTTP traffic: the communicator
```

Comment:
What is "the communicator"?

Reply:
Django-channels' Websockets consumer

https://channels.readthedocs.io/en/latest/topics/testing.html#websocketcommunicator

Oooh, actually it looks like "communicator" is the name of the test interface. I'll change that to "consumer".